src/simtools/simulator.py

```diff
@@ -109,43 +102,84 @@ def simulation_software(self, simulation_software):
             raise ValueError(f"Invalid simulation software: {simulation_software}")
         self._simulation_software = simulation_software.lower()
 
+    def _initialize_from_tool_configuration(self):
+        """Initialize simulator from tool configuration."""
+        self._test = self.args_dict.get("test", False)
+        self.sim_telarray_seeds = {
+            "seed": self.args_dict.get("sim_telarray_instrument_seeds"),
+            "random_instrument_instances": self.args_dict.get(
+                "sim_telarray_random_instrument_instances"
+            ),
+            "seed_file_name": "sim_telarray_instrument_seeds.txt",  # name only; no directory
+        }
+
+        if self.args_dict.get("corsika_file"):
```

Comment:
In case I forget to check later, is this mutually exclusive with the simulation software argument?